From 708c9643ae6892ba3bcbcdcd2ad6a26612d7645e Mon Sep 17 00:00:00 2001
From: Charles Ci-Jyun Wu <dminus@andestech.com>
Date: Wed, 1 Feb 2023 11:32:06 +0800
Subject: [PATCH 002/168] riscv: andes: defconfig: add andes_defconfig and
 andes-support.config

Andes defconfig generated XXX_defconfig using a make PHONY target.

sv32: ae350_rv32_[up/smp]_defconfig
sv39: ae350_rv64_[up/smp]_defconfig
sv48: ae350_rv64_[up/smp]_sv48_defconfig
AMP : ae350_rv64_ax25_amp_defconfig

Starting with RISCV-Linux-6.6, zram config is enabled by default.
https://gitea.andestech.com/RD-SW/linux/issues/140

Signed-off-by: Charles Ci-Jyun Wu <dminus@andestech.com>
---
 arch/riscv/Makefile                     |  28 ++++
 arch/riscv/configs/amp.config           |  12 ++
 arch/riscv/configs/andes-support.config |   2 +
 arch/riscv/configs/andes_defconfig      | 200 ++++++++++++++++++++++++
 arch/riscv/configs/smp.config           |   2 +
 arch/riscv/configs/sv39.config          |   2 +
 arch/riscv/configs/sv48.config          |   2 +
 7 files changed, 248 insertions(+)
 create mode 100644 arch/riscv/configs/amp.config
 create mode 100644 arch/riscv/configs/andes-support.config
 create mode 100644 arch/riscv/configs/andes_defconfig
 create mode 100644 arch/riscv/configs/smp.config
 create mode 100644 arch/riscv/configs/sv39.config
 create mode 100644 arch/riscv/configs/sv48.config

diff --git a/arch/riscv/Makefile b/arch/riscv/Makefile
index 4d06f3402674..4e2d92f1884d 100644
--- a/arch/riscv/Makefile
+++ b/arch/riscv/Makefile
@@ -188,3 +188,31 @@ rv32_defconfig:
 PHONY += rv32_nommu_virt_defconfig
 rv32_nommu_virt_defconfig:
 	$(Q)$(MAKE) -f $(srctree)/Makefile nommu_virt_defconfig 32-bit.config
+
+PHONY += ae350_rv32_up_defconfig
+ae350_rv32_up_defconfig:
+	$(Q)$(MAKE) -f $(srctree)/Makefile andes_defconfig andes-support.config 32-bit.config
+
+PHONY += ae350_rv32_smp_defconfig
+ae350_rv32_smp_defconfig:
+	$(Q)$(MAKE) -f $(srctree)/Makefile andes_defconfig andes-support.config 32-bit.config smp.config
+
+PHONY += ae350_rv64_up_defconfig
+ae350_rv64_up_defconfig:
+	$(Q)$(MAKE) -f $(srctree)/Makefile andes_defconfig andes-support.config 64-bit.config sv39.config
+
+PHONY += ae350_rv64_smp_defconfig
+ae350_rv64_smp_defconfig:
+	$(Q)$(MAKE) -f $(srctree)/Makefile andes_defconfig andes-support.config 64-bit.config smp.config sv39.config
+
+PHONY += ae350_rv64_up_sv48_defconfig
+ae350_rv64_up_sv48_defconfig:
+	$(Q)$(MAKE) -f $(srctree)/Makefile andes_defconfig andes-support.config 64-bit.config sv48.config
+
+PHONY += ae350_rv64_smp_sv48_defconfig
+ae350_rv64_smp_sv48_defconfig:
+	$(Q)$(MAKE) -f $(srctree)/Makefile andes_defconfig andes-support.config 64-bit.config smp.config sv48.config
+
+PHONY += ae350_rv64_ax25_amp_defconfig
+ae350_rv64_ax25_amp_defconfig:
+	$(Q)$(MAKE) -f $(srctree)/Makefile andes_defconfig andes-support.config 64-bit.config amp.config sv39.config
diff --git a/arch/riscv/configs/amp.config b/arch/riscv/configs/amp.config
new file mode 100644
index 000000000000..41ff91faefc6
--- /dev/null
+++ b/arch/riscv/configs/amp.config
@@ -0,0 +1,12 @@
+# CONFIG_ANDES_PPMA is not set
+CONFIG_AMP=y
+
+CONFIG_UIO=y
+CONFIG_UIO_PDRV_GENIRQ=y
+
+CONFIG_REMOTEPROC=y
+CONFIG_REMOTEPROC_CDEV=y
+CONFIG_ANDES_REMOTEPROC=m
+CONFIG_RPMSG_CHAR=m
+CONFIG_RPMSG_CTRL=m
+CONFIG_RPMSG_VIRTIO=m
diff --git a/arch/riscv/configs/andes-support.config b/arch/riscv/configs/andes-support.config
new file mode 100644
index 000000000000..376591a30097
--- /dev/null
+++ b/arch/riscv/configs/andes-support.config
@@ -0,0 +1,2 @@
+CONFIG_DEFAULT_HOSTNAME="andes-test"
+CONFIG_ARCH_ANDES=y
diff --git a/arch/riscv/configs/andes_defconfig b/arch/riscv/configs/andes_defconfig
new file mode 100644
index 000000000000..9135c4dd196a
--- /dev/null
+++ b/arch/riscv/configs/andes_defconfig
@@ -0,0 +1,200 @@
+CONFIG_SYSVIPC=y
+CONFIG_POSIX_MQUEUE=y
+CONFIG_NO_HZ_IDLE=y
+CONFIG_HZ_100=y
+CONFIG_HIGH_RES_TIMERS=y
+CONFIG_BPF_SYSCALL=y
+CONFIG_PREEMPT=y
+CONFIG_IKCONFIG=y
+CONFIG_IKCONFIG_PROC=y
+CONFIG_CGROUPS=y
+CONFIG_CGROUP_SCHED=y
+CONFIG_CFS_BANDWIDTH=y
+CONFIG_CGROUP_BPF=y
+CONFIG_NAMESPACES=y
+CONFIG_USER_NS=y
+CONFIG_CHECKPOINT_RESTORE=y
+CONFIG_BLK_DEV_INITRD=y
+CONFIG_INITRAMFS_SOURCE="rootfs-lite initramfs.txt.lite"
+# CONFIG_RD_BZIP2 is not set
+# CONFIG_RD_LZMA is not set
+# CONFIG_RD_XZ is not set
+# CONFIG_RD_LZO is not set
+# CONFIG_RD_LZ4 is not set
+# CONFIG_RD_ZSTD is not set
+# CONFIG_INITRAMFS_PRESERVE_MTIME is not set
+# CONFIG_SYSFS_SYSCALL is not set
+CONFIG_EMBEDDED=y
+CONFIG_PROFILING=y
+CONFIG_ARCH_ANDES=y
+CONFIG_SOC_VIRT=y
+# CONFIG_RISCV_ISA_ZICBOM is not set
+CONFIG_RISCV_SBI_V01=y
+CONFIG_PM=y
+CONFIG_CPU_IDLE=y
+CONFIG_CPU_FREQ=y
+CONFIG_JUMP_LABEL=y
+CONFIG_MODULES=y
+CONFIG_MODULE_UNLOAD=y
+CONFIG_NET=y
+CONFIG_PACKET=y
+CONFIG_UNIX=y
+CONFIG_NET_KEY=y
+CONFIG_INET=y
+# CONFIG_INET_DIAG is not set
+# CONFIG_WIRELESS is not set
+# CONFIG_ETHTOOL_NETLINK is not set
+CONFIG_DEVTMPFS=y
+CONFIG_DEVTMPFS_MOUNT=y
+CONFIG_BLK_DEV_LOOP=y
+CONFIG_VIRTIO_BLK=y
+CONFIG_SCSI=y
+CONFIG_BLK_DEV_SD=y
+CONFIG_BLK_DEV_SR=y
+CONFIG_SCSI_VIRTIO=y
+CONFIG_NETDEVICES=y
+# CONFIG_NET_VENDOR_ALACRITECH is not set
+# CONFIG_NET_VENDOR_AMAZON is not set
+# CONFIG_NET_VENDOR_AQUANTIA is not set
+# CONFIG_NET_VENDOR_ARC is not set
+# CONFIG_NET_VENDOR_ASIX is not set
+# CONFIG_NET_VENDOR_BROADCOM is not set
+# CONFIG_NET_VENDOR_CADENCE is not set
+# CONFIG_NET_VENDOR_CAVIUM is not set
+# CONFIG_NET_VENDOR_CORTINA is not set
+# CONFIG_NET_VENDOR_DAVICOM is not set
+# CONFIG_NET_VENDOR_ENGLEDER is not set
+# CONFIG_NET_VENDOR_EZCHIP is not set
+# CONFIG_NET_VENDOR_FUNGIBLE is not set
+# CONFIG_NET_VENDOR_GOOGLE is not set
+# CONFIG_NET_VENDOR_HUAWEI is not set
+# CONFIG_NET_VENDOR_INTEL is not set
+# CONFIG_NET_VENDOR_WANGXUN is not set
+# CONFIG_NET_VENDOR_LITEX is not set
+# CONFIG_NET_VENDOR_MELLANOX is not set
+# CONFIG_NET_VENDOR_MICREL is not set
+# CONFIG_NET_VENDOR_MICROCHIP is not set
+# CONFIG_NET_VENDOR_MICROSEMI is not set
+# CONFIG_NET_VENDOR_MICROSOFT is not set
+# CONFIG_NET_VENDOR_NI is not set
+# CONFIG_NET_VENDOR_NATSEMI is not set
+# CONFIG_NET_VENDOR_NETRONOME is not set
+# CONFIG_NET_VENDOR_PENSANDO is not set
+# CONFIG_NET_VENDOR_QUALCOMM is not set
+# CONFIG_NET_VENDOR_RENESAS is not set
+# CONFIG_NET_VENDOR_ROCKER is not set
+# CONFIG_NET_VENDOR_SAMSUNG is not set
+# CONFIG_NET_VENDOR_SEEQ is not set
+# CONFIG_NET_VENDOR_SOLARFLARE is not set
+# CONFIG_NET_VENDOR_SOCIONEXT is not set
+# CONFIG_NET_VENDOR_STMICRO is not set
+# CONFIG_NET_VENDOR_SYNOPSYS is not set
+# CONFIG_NET_VENDOR_VERTEXCOM is not set
+# CONFIG_NET_VENDOR_VIA is not set
+# CONFIG_NET_VENDOR_WIZNET is not set
+# CONFIG_NET_VENDOR_XILINX is not set
+# CONFIG_WLAN is not set
+CONFIG_INPUT_MOUSEDEV=y
+CONFIG_SERIAL_8250=y
+CONFIG_SERIAL_8250_CONSOLE=y
+CONFIG_SERIAL_OF_PLATFORM=y
+CONFIG_SERIAL_EARLYCON_RISCV_SBI=y
+CONFIG_VIRTIO_CONSOLE=y
+CONFIG_HW_RANDOM=y
+CONFIG_HW_RANDOM_VIRTIO=y
+CONFIG_I2C=y
+CONFIG_GPIOLIB=y
+CONFIG_DEBUG_GPIO=y
+CONFIG_GPIO_SYSFS=y
+CONFIG_GPIO_SIFIVE=y
+CONFIG_POWER_SUPPLY=y
+# CONFIG_HWMON is not set
+CONFIG_FB=y
+CONFIG_FRAMEBUFFER_CONSOLE=y
+# CONFIG_USB_SUPPORT is not set
+CONFIG_MMC=y
+CONFIG_MMC_SDHCI=y
+CONFIG_MMC_SDHCI_PLTFM=y
+CONFIG_MMC_SDHCI_CADENCE=y
+CONFIG_DMADEVICES=y
+CONFIG_SYNC_FILE=y
+CONFIG_VIRTIO_BALLOON=y
+CONFIG_VIRTIO_INPUT=y
+CONFIG_VIRTIO_MMIO=y
+# CONFIG_VHOST_MENU is not set
+# CONFIG_IOMMU_SUPPORT is not set
+CONFIG_EXT2_FS=y
+CONFIG_EXT4_FS=y
+CONFIG_EXT4_FS_POSIX_ACL=y
+CONFIG_EXT4_FS_SECURITY=y
+CONFIG_EXPORTFS_BLOCK_OPS=y
+CONFIG_FANOTIFY=y
+CONFIG_MSDOS_FS=y
+CONFIG_VFAT_FS=y
+CONFIG_TMPFS=y
+CONFIG_TMPFS_POSIX_ACL=y
+CONFIG_HUGETLBFS=y
+CONFIG_CONFIGFS_FS=y
+# CONFIG_EFIVAR_FS is not set
+# CONFIG_MISC_FILESYSTEMS is not set
+CONFIG_NFS_FS=y
+CONFIG_NFS_V3_ACL=y
+CONFIG_NFS_V4=y
+CONFIG_NFS_V4_1=y
+CONFIG_NFS_V4_2=y
+CONFIG_NFS_USE_LEGACY_DNS=y
+# CONFIG_NFS_DISABLE_UDP_SUPPORT is not set
+CONFIG_NLS_CODEPAGE_437=y
+CONFIG_NLS_ISO8859_1=y
+CONFIG_CRYPTO_DEV_VIRTIO=y
+CONFIG_CRC_ITU_T=y
+CONFIG_CRC7=y
+CONFIG_XZ_DEC=y
+CONFIG_PRINTK_TIME=y
+CONFIG_DEBUG_INFO_DWARF4=y
+CONFIG_GDB_SCRIPTS=y
+CONFIG_READABLE_ASM=y
+CONFIG_DEBUG_FS=y
+CONFIG_DEBUG_PAGEALLOC=y
+CONFIG_PTDUMP_DEBUGFS=y
+CONFIG_SCHED_STACK_END_CHECK=y
+CONFIG_DEBUG_VM=y
+CONFIG_DEBUG_VM_PGFLAGS=y
+CONFIG_DEBUG_MEMORY_INIT=y
+CONFIG_SOFTLOCKUP_DETECTOR=y
+CONFIG_WQ_WATCHDOG=y
+CONFIG_DEBUG_TIMEKEEPING=y
+# CONFIG_DEBUG_PREEMPT is not set
+CONFIG_DEBUG_RT_MUTEXES=y
+CONFIG_DEBUG_SPINLOCK=y
+CONFIG_DEBUG_MUTEXES=y
+CONFIG_DEBUG_RWSEMS=y
+CONFIG_DEBUG_LIST=y
+CONFIG_DEBUG_PLIST=y
+CONFIG_DEBUG_SG=y
+# CONFIG_RCU_TRACE is not set
+CONFIG_RCU_EQS_DEBUG=y
+# CONFIG_FTRACE is not set
+# CONFIG_RUNTIME_TESTING_MENU is not set
+CONFIG_MEMTEST=y
+
+CONFIG_SOUND=y
+CONFIG_SOUND_OSS_CORE=y
+CONFIG_SOUND_OSS_CORE_PRECLAIM=y
+CONFIG_SND=y
+CONFIG_SND_TIMER=y
+CONFIG_SND_PCM=y
+CONFIG_SND_OSSEMUL=y
+CONFIG_SND_PCM_OSS=y
+CONFIG_SND_PCM_OSS_PLUGINS=y
+CONFIG_SND_PCM_TIMER=y
+CONFIG_SND_SUPPORT_OLD_API=y
+CONFIG_SND_PROC_FS=y
+CONFIG_SND_VERBOSE_PROCFS=y
+CONFIG_SND_DRIVERS=y
+CONFIG_SND_HDA_PREALLOC_SIZE=64
+
+CONFIG_ZSMALLOC=y
+CONFIG_ZRAM=y
+CONFIG_CRYPTO_LZO=y
+CONFIG_SWAP=y
diff --git a/arch/riscv/configs/smp.config b/arch/riscv/configs/smp.config
new file mode 100644
index 000000000000..5d724af52886
--- /dev/null
+++ b/arch/riscv/configs/smp.config
@@ -0,0 +1,2 @@
+CONFIG_SMP=y
+CONFIG_HOTPLUG_CPU=y
diff --git a/arch/riscv/configs/sv39.config b/arch/riscv/configs/sv39.config
new file mode 100644
index 000000000000..bf107b886542
--- /dev/null
+++ b/arch/riscv/configs/sv39.config
@@ -0,0 +1,2 @@
+CONFIG_CMDLINE="no4lvl earlycon=sbi"
+CONFIG_CMDLINE_EXTEND=y
diff --git a/arch/riscv/configs/sv48.config b/arch/riscv/configs/sv48.config
new file mode 100644
index 000000000000..94136f06251d
--- /dev/null
+++ b/arch/riscv/configs/sv48.config
@@ -0,0 +1,2 @@
+CONFIG_CMDLINE="no5lvl earlycon=sbi"
+CONFIG_CMDLINE_EXTEND=y
-- 
2.34.1

